imports/ui/routes/methods.js

```diff
@@ -0,0 +1,22 @@
+const generateRoutes = (path, params) => {
+    // params is an object that consists of params for route 
+    const paramOccurances = path.match(/(:[A-z])\w+/g);
+    // check if all params that path needs are received
+    let paramsNeeded = [];
+    for (let i = 0; i < paramOccurances.length; i++) {
+        if (!(paramOccurances[i].substring(1) in params)) {
+            paramsNeeded.push(paramOccurances[i].substring(1));
+        }
+    }
+    if (paramsNeeded.length != 0) {
```

Comment:
You can leave just paramsNeeded.length. This is considered true for positive or negative numbers and false for 0